bolt-delegations-cli/src/main.rs

```diff
@@ -0,0 +1,145 @@
+                let signing_root = compute_commit_boost_signing_root(message.digest(), chain)?;
+                let signature = sk.sign(signing_root.0.as_ref());
+                let signed = SignedDelegation { message, signature };
+                signed_messages.push(SignedMessage::Delegation(signed))
+            }
+            Action::Revoke => {
+                let message = RevocationMessage::new(sk.public_key(), delegatee_pubkey.clone());
+                let signing_root = compute_commit_boost_signing_root(message.digest(), chain)?;
+                let signature = sk.sign(signing_root.0.as_ref());
+                let signed = SignedRevocation { message, signature };
+                signed_messages.push(SignedMessage::Revocation(signed));
+            }
+        }
+    }
+
+    Ok(signed_messages)
+}
+
+/// Generate signed delegations/revocations using a keystore file
+///
+/// - Read the keystore file
+/// - Decrypt the keypairs using the password
+/// - Create messages
+/// - Compute the signing roots and sign the message
+/// - Return the signed message
+fn generate_from_keystore(
+    keys_path: &str,
+    password: &str,
+    delegatee_pubkey: BlsPublicKey,
+    chain: &Chain,
+    action: Action,
+) -> Result<Vec<SignedMessage>> {
+    let keystores_paths = keystore_paths(keys_path)?;
+    let mut signed_messages = Vec::with_capacity(keystores_paths.len());
+
+    for path in keystores_paths {
+        let kp = Keystore::from_json_file(path.clone()).map_err(KeystoreError::Eth2Keystore)?;
+        let kp = kp.decrypt_keypair(password.as_bytes()).map_err(KeystoreError::Eth2Keystore)?;
+        let validator_pubkey = BlsPublicKey::try_from(kp.pk.to_string().as_ref())?;
+        let validator_private_key = kp.sk;
+
+        match action {
+            Action::Delegate => {
+                let message = DelegationMessage::new(validator_pubkey, delegatee_pubkey.clone());
+                let signing_root = compute_commit_boost_signing_root(message.digest(), chain)?;
+                let signature = validator_private_key.sign(signing_root.0.into());
+                let signature = BlsSignature::try_from(signature.serialize().as_ref())?;
+                let signed = SignedDelegation { message, signature };
+                signed_messages.push(SignedMessage::Delegation(signed));
+            }
+            Action::Revoke => {
+                let message = RevocationMessage::new(validator_pubkey, delegatee_pubkey.clone());
+                let signing_root = compute_commit_boost_signing_root(message.digest(), chain)?;
+                let signature = validator_private_key.sign(signing_root.0.into());
+                let signature = BlsSignature::try_from(signature.serialize().as_ref())?;
+                let signed = SignedRevocation { message, signature };
+                signed_messages.push(SignedMessage::Revocation(signed));
+            }
+        }
```

Comment:
Can be simplified in a similar fashion as above

Reply:
same as above, it can't